### ISSUE TITLE
Conditional transition?

### ISSUE THREAD

**Issue by jesperlandberg** [Dogstudio/highway#5]
Hi again,

This is not an issue, more of a question on how this would be done in the most effecient way.

I know one can set different transitions for different renderers, but is it possible to change transition depending on other stuff, like how would I do something like this:

```
transitions: {
  'default': document.body.classList.contains('is-open') : TransitionOne : TransitionTwo
}
```
I guess I could just do my conditional stuff inside the transitions file but separating them into two transitions would be cleaner.

**Comment by Anthodpnt** [comment]
Hi @jesperlandberg,

Yes you'll have to do your conditional stuffs in your transitions.
The `Highway.Core` is called once to setup Highway with the renderers and transitions.

This means that once it's called it's not aware of any DOM changes.
If you want to watch DOM changes and change the transition you'll have to work within the transitions.

This way when the transition is called, if the `body` has the classname `is-open` you can change it.
I'm curious, what kind of transition would you like to do ?

Because if it's an animation when a menu or something is opened this shouldn't be part of the transitions of Highway but more a transition in a file like `header.js` for example that would be called when the menu button is triggered... Assuming this is the kind of animation you want to achieve.

So in conclusion, you can't do this kind of conditional statement when calling `Highway.Core` and I don't think this is a feature coming in next version since I don't think it's an intended behaviour for this kind of library.

Best regards,
Anthodpnt

**Comment by jesperlandberg** [comment]
Hi @Anthodpnt 

I understand:) no problem, will do my check in the transition.js file!

Well basically, I have a mask element that I animate on transition, but when the menu is open, a menu that is a full size overlay kind of menu, I will use the menu element as the mask, for the out part of the transition. No big deal and is easily made with conditionals inside the transition file. I attached a video clip if ur interested in seeing what I mean. Another option is just giving the default mask transition a delay and triggering my close menu method on NAVIGATE_OUT, so that the mask animation starts after the menu is closed... but that makes the whole switching pages to long=)

[transition.zip](https://github.com/Dogstudio/highway/files/2055279/transition.zip)

Loving the library so far btw! Good job:)

**Comment by Anthodpnt** [comment]
Really nice work @jesperlandberg !
I would like to share this site when it will be live to show what is possible with Highway if you agree :).

**Comment by jesperlandberg** [comment]
@Anthodpnt of course:) Will send over a link when live!:)

**Comment by Anthodpnt** [comment]
@jesperlandberg Thanks ! Closing the issue.

**Comment by jesperlandberg** [comment]
hi again @Anthodpnt .

So I'm trying to do some conditional checks inside the out() part of the transition, but it seems I cannot check for dynamically added classes? Example, the below doesn't work, even tho the element actually has the class "is-active". The console.log() just logs the initial render of the element, when it didn't have the "is-active" class.

```
  out(view, done) {
    console.log(document.querySelector('.js-toggle-menu'))

    if (btn.classList.contains('is-active')) {

      ...

    } else {

      ...
      
    }
  }
```
